supervisor/homeassistant/core.py

```diff
@@ -402,6 +402,15 @@ async def _block_till_run(self, version: AwesomeVersion) -> None:
             # 2: Check if API response
             if await self.sys_homeassistant.api.check_api_state():
                 _LOGGER.info("Detect a running Home Assistant instance")
+                with suppress(HomeAssistantError):
+                    async with self.sys_homeassistant.api.make_request(
+                        "get", "api/config"
+                    ) as resp:
+                        data = await resp.json()
+                        integrations = data.get("components", [])
+                        if "frontend" not in integrations:
+                            _LOGGER.error("API responded but frontend is not loaded")
+                            break
```

Comment:
Let's create a helper on api.py which we can also use for USB detection on first step. After that we should just use that on the update flow which is the only part they are interested if the frontend is up for the rollback

Reply:
USB detection uses websockets